### PR TITLE
Add TLS based tests to functional test suite

### LINKED ISSUE #3020: [distributed] Enable SSL based tests as well for all handler/controller/rpc/api communication
We should use a self-signed certificate and validate it.

### PR DESCRIPTION
**Pull request by krisis**
## Description

Add TLS based functional tests for minio server.
## Motivation and Context

The existing tests only test HTTP API requests. See #3020 
## How Has This Been Tested?

By running, `go test -check.v` and will be run as part of CI on every commit.
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

Fixes #3020

### CODE REVIEW

**Comment by krisis** [comment]
The addition of TLS based functional tests has increased the execution time of `make test`. As part of review, do suggest tests that can be removed.

File: test-data/server.key [hunk 1]
```diff
@@ -0,0 +1,27 @@
+-----BEGIN RSA PRIVATE KEY-----
```
Comment (harshavardhana) [inline test-data/server.key]: You don't need to keep this in a file you can just directly keep this in file as a global []byte array, avoids directories like this.

**Review by harshavardhana — COMMENTED** [review]
> The addition of TLS based functional tests has increased the execution time of make test. As part of review, do suggest tests that can be removed.

How substantial is the increase?

**Comment by harshavardhana** [comment]
> The addition of TLS based functional tests has increased the execution time of make test. As part of review, do suggest tests that can be removed.
> How substantial is the increase?

`go test -race` doesn't show any substantial increase, btw you don't need to copy all the tests. You can configure the Suite to have TLS and Non - TLS. 

```
// Init and run test on FS backend.
var _ = Suite(&TestSuiteCommon{serverType: "FS"})

// Init and run test on XL backend.
var _ = Suite(&TestSuiteCommon{serverType: "XL"})
```

Like how it is done here..

**Pull request merged**